monitoring/notification/alerts.yaml

```diff
@@ -0,0 +1,105 @@
+x-inputs:
+- name: namespace
+  type: constant
+  value: zenko
+- name: job_notification_producer
+  type: config
+  value: artesca-data-backbeat-notification-producer-headless
+- name: job_notification_processors
+  type: config
+  value: artesca-data-backbeat-notification-processor
+- name: configured_notification_targets
+  type: config
+- name: bucket_notification_enabled
+  type: config
+  value: 0
+- name: notificationFailuresWarningThreshold
+  type: config
+  value: 0.1
+- name: notificationFailuresCriticalThreshold
+  type: config
+  value: 1
+- name: notification_producer_replicas
+  type: config
+  value: 1
+- name: notification_processor_replicas
+  type: config
+  value: 1
+
+groups:
+- name: Bucket Notification
+  rules:
+
+  - alert: NotificationProducerDegraded
+    Expr: |
+      ${bucket_notification_enabled} == bool 1
+        * (sum(up{namespace="${namespace}",job="${job_notification_producer}"})
+        < ${notification_producer_replicas})
+    For: "30s"
+    Labels:
+     severity: warning
+    Annotations:
+      zenko_service: backbeat-notification-producer
+      description: "Less than 100% of notification producers are up and healthy"
+      summary: "Notification producer service is degraded"
+
+  - alert: NotificationProducerCritical
+    Expr: |
+      ${bucket_notification_enabled} == bool 1
+        * (sum(up{namespace="${namespace}",job="${job_notification_producer}"}) * 2
+        < ${notification_producer_replicas})
+    For: "30s"
+    Labels:
+      severity: critical
+    Annotations:
+      zenko_service: backbeat-notification-producer
+      description: "Less than 50% of notification producers are up and healthy"
+      summary: "Notification producer service is in critical state"
+
+  - alert: NotificationProcessorsDegraded
+    Expr: |
+      ${bucket_notification_enabled} == bool 1 
+        * (sum(up{namespace="${namespace}",job=~"${job_notification_processors}.*"}) 
+        < (${configured_notification_targets} * ${notification_processor_replicas}))
+    For: "30s"
+    Labels:
+      severity: warning
+    Annotations:
+      zenko_service: backbeat-notification-processor
+      description: "Less than 100% of notification processors are up and healthy"
+      summary: "Notification processor service is degraded"
+
+  - alert: NotificationProcessorsCritical
+    Expr: |
+      ${bucket_notification_enabled} == bool 1
+        * (sum(up{namespace="${namespace}",job=~"${job_notification_processors}.*"}) * 2 
+        < (${configured_notification_targets} * ${notification_processor_replicas}))
+    For: "30s"
+    Labels:
+      severity: critical
+    Annotations:
+      zenko_service: backbeat-notification-processor
+      description: "Less than 50% of notification processors are up and healthy"
+      summary: "Notification processor service is in critical state"
+
+  - alert: NotificationDeliveryDegraded
```

Comment:
for these it may make sense to have alerts for each target, what do you think?
since each target (or remote server, more precisely) may have different error rate... but then we cannot specify different warning thresholds, so...

Reply:
Yes i do think alerts should be precise and indicate exactly where the issue is, especially here since the targets might no be linked to each other and can be different machines.